File: mine-client.go
package main

import (
	"fmt"
	"net"
)

func main() {
	conn, err := net.Dial("tcp", "127.0.0.1:8080")
	if err != nil {
		fmt.Printf("Error connecting to server: %s\n", err)
		return
	}
	defer conn.Close()

	fmt.Println("Connected to server")

	// Send data to the server
	nonce := "123456"
	_, err = conn.Write([]byte(nonce))
	if err != nil {
		fmt.Printf("Error writing to server: %s\n", err)
		return
	}

	// Read hash from the server
	buffer := make([]byte, 1024)
	n, err := conn.Read(buffer)
	if err != nil {
		fmt.Printf("Error reading from server: %s\n", err)
		return
	}
	hash := string(buffer[:n])
	fmt.Printf("Received hash from server: %s\n", hash)
}

File: mine-client.rs
use tokio::net::TcpStream;
use std::io::{self, Read, Write};
use std::net::SocketAddr;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let addr = "127.0.0.1:8080".parse::<SocketAddr>()?;
    let mut stream = TcpStream::connect(&addr).await?;
    println!("Connected to server");

    // Send data to the server
    let nonce = "123456";
    stream.write_all(nonce.as_bytes()).await?;
    stream.flush().await?;

    // Read hash from the server
    let mut buffer = [0; 1024];
    let bytes_read = stream.read(&mut buffer).await?;
    let hash = String::from_utf8_lossy(&buffer[..bytes_read]);
    println!("Received hash from server: {:?}", hash);

    Ok(())
}
